website/content/en/docs/concepts/nodeclasses.md

```diff
@@ -1528,6 +1532,142 @@ If a `NodeClaim` requests `vpc.amazonaws.com/efa` resources, `spec.associatePubl
 requires that the field is only set to true when configuring an instance with a single ENI at launch. When using this field, it is advised that users segregate their EFA workload to use a separate `NodePool` / `EC2NodeClass` pair.
 {{% /alert %}}
 
+## spec.enclaveOptions
+
+The `enclaveOptions` field allows you to specify whether the instance is enabled for Amazon Web Services Nitro Enclaves. Nitro Enclaves provide an isolated compute environment to protect and process highly sensitive data.
+
+To use Nitro Enclaves, you must use the Custom AMI family to remove system resources for the enclave. Currently, this can only be achieved with the Custom AMI family combined with systemReserve to remove resources, since the Custom AMI family doesn't translate the system reserve to the underlying node but can remove the resources from Karpenter's scheduling simulation.
+
+### What are Nitro Enclaves?
```

Comment:
Let's add a separate task section for the enclaves, so that we can describe the configuration that is need for setting up HugePages with NodeOverlay and the userData.